knowledge/knowledge_read.go

```diff
@@ -255,6 +260,56 @@ func parseMultipleNodesFromPaths(paths []*knowledgepb.Path) ([]*knowledgepb.Node
 	return nodes, nil
 }
 
+func GenerateRandomString(length int) string {
+	b := make([]byte, length)
+	_, err := rand.Read(b)
+	if err != nil {
+		panic(err)
+	}
+	return base64.StdEncoding.EncodeToString(b)
+}
+
+// NodesRecordsWithTypeNode is a helper function that get all nodes of specific type either DigitalTwin
+// or Resource and transform them into records.
+func (c *Client) NodesRecordsWithTypeNode(
+	ctx context.Context,
+	nodeType string,
+	opts ...grpc.CallOption,
+) ([]*ingestpb.Record, error) {
+	ctx = insertMetadata(ctx, c.xMetadata)
+	path := fmt.Sprintf("(:%s)", nodeType)
+	resp, err := c.client.IdentityKnowledge(ctx, &knowledgepb.IdentityKnowledgeRequest{
+		Path:      path,
+		Operation: knowledgepb.Operation_OPERATION_READ,
+	}, opts...)
+	if err != nil {
+		return nil, err
+	}
+	nodes, err := parseMultipleNodesFromPaths(resp.GetPaths())
+	if err != nil {
+		return nil, err
+	}
+	records := []*ingestpb.Record{}
+	for _, node := range nodes {
+		caser := cases.Title(language.English)
+		record := &ingestpb.Record{
+			Id: GenerateRandomString(12),
+			Operation: &ingestpb.Record_Delete{
+				Delete: &ingestpb.DeleteData{
+					Data: &ingestpb.DeleteData_Node{
+						Node: &ingestpb.NodeMatch{
+							ExternalId: node.ExternalId,
+							Type:       caser.String(node.Type),
+						},
+					},
+				},
+			}, // lint:file-ignore U1000 Ignore report
+		}
+		records = append(records, record)
+	}
+	return records, nil
+}
```

Comment:
Can you explain what is the purpose of this method? Why it is generating some random string? This sounds kind a weird. @maaland any inputs?

Reply:
the random is to generate an id for for record: they can be anything, their value is not used for anything else

Reply:
the purpose is to get all nodes of a certain type and transform them into records ready to be ingested

Reply:
after discussion with @maaland, I'm going to create a helper package